include/caffe/util/db.hpp

```diff
@@ -0,0 +1,188 @@
+#ifndef CAFFE_UTIL_DB_HPP
+#define CAFFE_UTIL_DB_HPP
+
+#include <string>
+
+#include "leveldb/db.h"
+#include "leveldb/write_batch.h"
+#include "lmdb.h"
+
+#include "caffe/common.hpp"
+#include "caffe/proto/caffe.pb.h"
+
+namespace caffe { namespace db {
+
+enum Mode { READ, WRITE, NEW };
+
+class Cursor {
+ public:
+  Cursor() { }
+  virtual ~Cursor() { }
+  virtual void SeekToFirst() = 0;
+  virtual void Next() = 0;
+  virtual string key() = 0;
+  virtual string value() = 0;
+  virtual bool valid() = 0;
```

Comment:
I went with `valid` since it's an accessor and we generally lowercase those, but presumably LevelDB is also following the Google C++ style guide, so I could see the argument either way.